src/Microsoft.Identity.Client/ITokenCacheInternal.cs

```diff
@@ -60,5 +62,14 @@ internal interface ITokenCacheInternal : ITokenCache, ITokenCacheSerializer
         void ClearAdalCache();
         void ClearMsalCache();
         Task ClearAsync();
+
+        #endregion
+
+        #region Cache notifications
+        Task OnAfterAccessAsync(TokenCacheNotificationArgs args);
```

Comment:
Although CacheSessionManager is responsible for keeping the cache in sync, our public API allows users to configure delegates on the TokenCache object.